src/utils/group.ts

```diff
@@ -20,51 +20,70 @@
  */
 
 import { assert } from "../util";
+import { View, ViewMode } from "../components/editor/View";
 import { File } from "../model";
 
 export default class Group {
-  file: File;
-  files: File[];
-  preview: File;
-  constructor(file: File, preview: File, files: File[]) {
-    this.file = file;
+  view: View;
+  views: View[];
+  preview: View;
```

Comment:
let's make it boolean `isPreview` -- I don't see the case when `this.view !== this.preview`

Reply:
`preview` references to the view which is currently open in preview mode. So it should just be a View.

Reply:
but it's essentially the same is the `currentView` it just specifies that `currentView` is in preview mode, is it correct? If yes, I think having a comment about that or name change is nice to have.

Reply:
Nope, `currentView` means the view currently being displayed in the pane, and `preview` means the view which is in being previewed (The tab which is shown in italics)

Reply:
I'm looking at the code and I see that every time the `currentView` is assigned, the `preview` is assigned to the same exact value or null. Is it correct? If it is correct invariant, we shall enforce it.

Btw, the `close()` is missing such assignment, and that probably a bug since this.preview might stay assigned to the closed view.

Reply:
Yes, preview and currentView are same when we open new files, but in case of switching between the views, you can see that these two are in fact different, in that case only `currentView` changes.

And yes, `close()` is missing this, I'll update